src/Catel.MVVM/Catel.MVVM.Shared/Collections/FastObservableCollection.cs

```diff
@@ -326,6 +422,112 @@ protected override void OnPropertyChanged(PropertyChangedEventArgs e)
+            }
+
+            // Call base
+            base.ClearItems();
+        }
+
+        /// <summary>
+        /// Inserts an item into the collection at the specified index.
+        /// </summary>
+        /// <param name="index">The zero-based index at which <paramref name="item"/> should be inserted.</param><param name="item">The object to insert.</param>
+        protected override void InsertItem(int index, T item)
+        {
+            // Check
+            if (_suspensionContext != null && _suspensionContext.Mode == SuspensionMode.Removing)
+            {
+                throw Log.ErrorAndCreateException<InvalidOperationException>("Adding items is not allowed in removing mode.");
+            }
+
+            // Call base
+            base.InsertItem(index, item);
+
+            if (_suspensionContext != null && _suspensionContext.Mode == SuspensionMode.Adding)
+            {
+                // Remember
+                _suspensionContext.NewItems.Add(item);
+                _suspensionContext.NewItemIndices.Add(index);
+            }
+        }
+
+        /// <summary>
+        /// Moves the item at the specified index to a new location in the collection.
+        /// </summary>
+        /// <param name="oldIndex">The zero-based index specifying the location of the item to be moved.</param><param name="newIndex">The zero-based index specifying the new location of the item.</param>
+        protected override void MoveItem(int oldIndex, int newIndex)
+        {
+            // Check
+            if (_suspensionContext != null && _suspensionContext.Mode != SuspensionMode.None)
+            {
+                throw Log.ErrorAndCreateException<InvalidOperationException>("Moving items is not allowed in modes other than mixed.");
+            }
+
+            // Call base
+            base.MoveItem(oldIndex, newIndex);
+        }
+
+        /// <summary>
+        /// Removes the item at the specified index of the collection.
+        /// </summary>
+        /// <param name="index">The zero-based index of the element to remove.</param>
+        protected override void RemoveItem(int index)
+        {
+            // Check
+            if (_suspensionContext != null && _suspensionContext.Mode == SuspensionMode.Adding)
+            {
+                throw Log.ErrorAndCreateException<InvalidOperationException>("Removing items is not allowed in adding mode.");
+            }
+
+            // Get item
+            T item;
+            if (_suspensionContext != null && _suspensionContext.Mode == SuspensionMode.Removing)
+            {
+                item = this[index];
+            }
+            else
+            {
+                item = default(T);
+            }
+
+            // Call base
+            base.RemoveItem(index);
+
+            if (_suspensionContext != null && _suspensionContext.Mode == SuspensionMode.Removing)
```

Comment:
Remember: Mixed is now None and None ends in a Reset event with "empty" event args. Still questionable?